File: 16066.cpp
#include <iostream>

using namespace std;

long long gcd(long long a, long long b){
	if(b == 0){
		return a;
	}else{
		return gcd(b, a % b);
	}
}

bool prime(long long n){
	if(n == 1){
		return false;
	}
	for(long long p = 2; p * p <= n; ++p){
		if(n % p == 0){
			return false;
		}
	}
	return true;
}

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	int n;
	cin >> n;
	for(int i = 0; i < n; ++i){
		long double a, b;
		cin >> a >> b;
		long long x = (long long)(a * 100000), y = (long long)(b * 100000);
		long long g = gcd(x, y);
		x /= g;
		y /= g;
		if(x == 1 && y == 1){
			x = 2;
			y = 2;
		}
		if(prime(x) && prime(y)){
			cout << x << ' ' << y << '\n';
		}else{
			cout << "impossible\n";
		}
	}
	return 0;
}

File: 16066.py
import sys
import math
import decimal


def prime(n):
    if n == 1:
        return False
    for p in range(2, int(n ** 0.5) + 1):
        if n % p == 0:
            return False
    return True


def main():
    decimal.getcontext().prec = 7
    t = int(sys.stdin.readline())
    for _ in range(t):
        a, b = (int(decimal.Decimal(x) * 100000) for x in sys.stdin.readline().split())
        g = math.gcd(a, b)
        a //= g
        b //= g
        if a == 1 and b == 1:
            a *= 2
            b *= 2
        if prime(a) and prime(b):
            print(a, b)
        else:
            print("impossible")


main()
